### PR TITLE
Create clinical document service and refit API for AB#13792

### PR DESCRIPTION
**Pull request by rayking29**
# Fixes or Implements [AB#13792](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13792)

## Description

- Update clinical doc service to retrieve records
- create clinical document API with refit method to get records
- add backend model 

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs [hunk 1]
```diff
@@ -26,13 +26,13 @@ public class ClinicalDocumentRecord
         /// <summary>
         /// Gets or sets the ID.
```
Comment (BrianMaki) [inline Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs:27]: Nit: Update documentation to reflect change.

File: Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs [hunk 2]
```diff
@@ -26,13 +26,13 @@ public class ClinicalDocumentRecord
...
         /// </summary>
-        [JsonPropertyName("id")]
+        [JsonPropertyName("healthDataId")]
         public string Id { get; set; } = string.Empty;
 
         /// <summary>
         /// Gets or sets the file ID.
```
Comment (BrianMaki) [inline Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs:33]: Nit: Update documentation to reflect change.

**Review by BrianMaki — COMMENTED** [review]
A couple of nits.

**Review by BrianMaki — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

File: Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs [hunk 3]
```diff
@@ -26,13 +26,13 @@ public class ClinicalDocumentRecord
...
         /// </summary>
-        [JsonPropertyName("fileId")]
+        [JsonPropertyName("healthDataFileId")]
```
Comment (MikeLyttle) [inline Apps/ClinicalDocument/src/Models/ClinicalDocumentRecord.cs:35]: This is the model that gets sent to our front-end.  I thought we should have a separate model for the data returned by PHSA and use a mapper to convert them in the service.  Can discuss after stand-up.

**Pull request merged**